streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -3474,14 +3453,13 @@ public void shouldNotCommitActiveAndStandbyTasksWhileRebalanceInProgress() throw
         final StateMachineTask task01 = new StateMachineTask(taskId01, taskId01Partitions, false);
 
         makeTaskFolders(taskId00.toString(), task01.toString());
-        expectLockObtainedFor(taskId00, taskId01);
```

Comment:
This was reported by Mockito as an unnecessary stubbing. I checked the code and it appears we never call this as part of the test. Let me know in case I am wrong.

Reply:
That is because there is a bug in the test.

Line
```java
makeTaskFolders(taskId00.toString(), task01.toString());
```
should be
```java
makeTaskFolders(taskId00.toString(), taskId01.toString());
```

Variable `task01` holds a `StateMachineTask` and not a `TaskId`. Consequently, `task01.toString()` returns the string representation of a `StateMachineTask` and not of a `TaskId` which leads to a `TaskIdFormatException` in this [call](https://github.com/apache/kafka/blob/6fe74f78dc725466319bd0465764913532949ba3/streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java#L1184) and skips the lock.

Please fix the bug and readd the stub.